eos-phone-home

```diff
@@ -148,6 +153,43 @@ class PhoneHome(object):
 
         return None
 
+    def _get_metrics(self):
+        permissions_file = '/etc/metrics/eos-metrics-permissions.conf'
+        valid_environments = ['production', 'dev', 'test']
+
+        enabled = False
+        uploading_enabled = True
+        environment = 'production'
+
+        try:
+            config = configparser.ConfigParser()
+            config.read(permissions_file)
+            enabled = config['global'].getboolean('enabled', enabled)
+            uploading_enabled = config['global'].getboolean('uploading_enabled',
+                uploading_enabled)
+            environment = config['global'].get('environment', environment)
+        except Exception as e:
+            print("Unable to read %s, assuming fallback values." %
+                (permissions_file, ), e, file=sys.stderr)
+
+        # the event recorder replaces invalid environments with 'test'
+        if environment not in valid_environments:
+            print('Found invalid metrics environment', environment,
+                file=sys.stderr)
+            environment = 'test'
+
+        return (enabled and uploading_enabled, environment)
+
+    def _get_metrics_enabled(self):
+        enabled = self._lookup_or_get_variable('metrics')[0]
```

Comment:
Yeah - this will happen... we read it every time we sent the ping event, and it's included in ping rather than activate - because we know it can change - so we'll have to do some kind of time-boxed analysis (eg events for the last day, compare unique systems on both ping and metrics). We also need to be aware that if count=0, if the user has a wired connection the event could've been sent with the metrics configuration from before the user passes the FBE and expresses a preference about metrics at all. So if we want to do opt-in/opt-out ratio analysis, we should look at count >= 1.